pkg/datapath/linux/bandwidth/ops.go

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package bandwidth
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cilium/cilium/pkg/datapath/tables"
+	"github.com/cilium/cilium/pkg/datapath/types"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/reconciler"
+	"github.com/sirupsen/logrus"
+	"github.com/vishvananda/netlink"
+)
+
+type ops struct {
+	log       logrus.FieldLogger
+	isEnabled func() bool
+}
+
+func newOps(log logrus.FieldLogger, mgr types.BandwidthManager) reconciler.Operations[*tables.DesiredQdisc] {
+	return &ops{log, mgr.Enabled}
+}
+
+// Delete implements reconciler.Operations.
+func (*ops) Delete(context.Context, statedb.ReadTxn, *tables.DesiredQdisc) error {
+	// We don't restore qdiscs.
+	return nil
+}
+
+// Prune implements reconciler.Operations.
+func (*ops) Prune(context.Context, statedb.ReadTxn, statedb.Iterator[*tables.DesiredQdisc]) error {
+	// We don't touch qdiscs of other devices.
+	return nil
+}
+
+// Update implements reconciler.Operations.
+func (ops *ops) Update(ctx context.Context, txn statedb.ReadTxn, q *tables.DesiredQdisc) (changed bool, err error) {
+	if !ops.isEnabled() {
```

Comment:
Kind of hacky. I didn't think of needing to register the reconciler's lifecycle hooks based on probing done from a start hook. One solution is to have a lifecycle internal to just the module to which this gets registered (like we kind do in the operator with the leader lifecycle). Another would be to not have e.g. `reconciler.New` register it's lifecycle hooks and we'd call it's `Start` hook after probe says go.

Though even better would be if we would just bail out hard if `EnableBandwidthManager=true` and probing says it's not supported...